### PR TITLE
[plugin.video.wakanim] v0.3.0

### PR DESCRIPTION
**Pull request by MrKrabat**
### Description
Wakanim.tv is a paid anime streaming service and is offering subtitles for three languages (english, german, french).
Wakanim is available at this countries: Germany, France, Belgium, Switzerland, Quebec, Sweden, Norway, Iceland, Finland, Denmark.
A paid subscription costs 5€/month, otherwise it is possible to buy anime for 1€/episode which allows downloading. [currently we only support streaming with subscription].

I am in no way affiliated with Wakanim.tv.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.wakanim/addon.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.wakanim" name="Wakanim" version="0.3.0" provider-name="MrKrabat">
+    <requires>
+        <import addon="xbmc.python" version="2.24.0"/>
```
Comment (romanvm) [inline plugin.video.wakanim/addon.xml]: It's 2.25 for Krypton

File: plugin.video.wakanim/default.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import xbmc
+import xbmcaddon
+
+
+_plugId = "plugin.video.wakanim"
+
+# plugin constants
+__addon__   = xbmcaddon.Addon(id=_plugId)
+__plugin__  = __addon__.getAddonInfo("name")
+__version__ = __addon__.getAddonInfo("version")
```
Comment (romanvm) [inline plugin.video.wakanim/default.py]: It is not recommended to use names with double underscores for custom variables. Those are reserved for Python internal stuff.

File: plugin.video.wakanim/resources/lib/wakanim.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    if args._country == "0":
+        args._country = "de"
+    elif args._country == "1":
+        args._country = "fr"
+    elif args._country == "2":
+        args._country = "sc"
+    else:
+        args._country = "de"
+
+    if (username == "") or (password == ""):
```
Comment (romanvm) [inline plugin.video.wakanim/resources/lib/wakanim.py]: You don't test for an empty string such way. An empty string evaluates to `False` by itself. It should be
```python
if not (username and password):
    ...
```

File: plugin.video.wakanim/resources/lib/wakanim.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        # open addon settings
+        args._addon.openSettings()
+        return False
+    else:
+        # login
+        success = login.login(username, password, args)
+        if success == True:
```
Comment (romanvm) [inline plugin.video.wakanim/resources/lib/wakanim.py]: The same. `if success:` is enough.

File: plugin.video.wakanim/resources/lib/wakanim.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            xbmcgui.Dialog().ok(args._addonname, args._addon.getLocalizedString(30040))
+            return False
+
+
+def check_mode(args):
+    """Run mode-specific functions
+    """
+    try:
+        mode = args.mode
+    except:
```
Comment (romanvm) [inline plugin.video.wakanim/resources/lib/wakanim.py]: Don't use bare `except` unless you really know what you are doing. You should test against specific exceptions.

File: plugin.video.wakanim/resources/lib/list.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                          thumbnailImage = info["thumb"])
+    li.setInfo(type       = mediatype,
+               infoLabels = {"title":   info["title"],
+                             "plot":    info["plot"],
+                             "year":    info["year"],
+                             "episode": info["episode"],
+                             "season":  info["season"],
+                             "rating":  info["rating"],
+                             "genre":   "Anime"})
+    li.setProperty("Fanart_Image", info["fanart_image"])
```
Comment (romanvm) [inline plugin.video.wakanim/resources/lib/list.py]: Use `li.setArt()` to set all types of images for the list item: icon, thumb, fanart and others.

File: plugin.video.wakanim/resources/lib/list.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    # Defaults in dict. Use "None" instead of None so it is compatible for
+    info = set_info_defaults(args, info)
+    u = build_url(info)
+
+    # Create list item
+    li = xbmcgui.ListItem(label          = info["title"],
+                          iconImage      = info["thumb"],
+                          thumbnailImage = info["thumb"])
+    li.setInfo(type       = mediatype,
+               infoLabels = {"title":   info["title"],
```
Comment (romanvm) [inline plugin.video.wakanim/resources/lib/list.py]: It is recommended to set `'mediatype'` infolabel as well.

File: plugin.video.wakanim/resources/lib/list.py [hunk 8]
```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
```
Comment (romanvm) [inline plugin.video.wakanim/resources/lib/list.py]: **Don't** ever use names that shadow Python built-in names like `list` and such! Please change the name of the module.

File: plugin.video.wakanim/resources/lib/netapi.py [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+import re
+import urllib
+import urllib2
+from bs4 import BeautifulSoup
+
+import xbmc
+import xbmcgui
+
+import login
+import list
```
Comment (romanvm) [inline plugin.video.wakanim/resources/lib/netapi.py]: Invalid module name, see my comment above.

**Comment by MrKrabat** [comment]
Thank you for taking your time to find so many improvements.

I have updated the commit

**Comment by MrKrabat** [comment]
Sorry, I updated the commit again.
mediatype Infolabel is only set if it is not a folder.

**Comment by romanvm** [comment]
LGTM

**Pull request merged**